zeeguu_api/api/dashboard.py

```diff
@@ -0,0 +1,297 @@
+from flask import jsonify
+import json
+import random
+import datetime
+
+db = zeeguu.db
+
+
+# Checked to see if the user requesting has permission to view the class with id 'class_id'
+def _has_permission_for_class(class_id):
+    from zeeguu.model import TeacherCohortMap
+    link = TeacherCohortMap.query.filter_by(cohort_id=class_id).all()
+    for l in link:
+        if l.user_id == flask.g.user.id:
+            return True
+    return False
+
+
+# Checks to see if user has a valid active session
+@api.route("/has_session", methods=["GET"])
+def has_session():
+    try:
+        session_id = int(flask.request.args['session'])
+        session = Session.query.filter_by(id=session_id).one()
+        return jsonify(1)
+    except:
+        return jsonify(0)
+
+
+@api.route("/get_class_permissions/<id>", methods=["GET"])
+@with_session
+def get_class_permissions(id):
+    """
+
+        Checks to see if user has permissions to access a certain class.
+
+    """
+    if (_has_permission_for_class(id)):
+        return jsonify(1)
+    return jsonify(0)
+
+
+@api.route("/get_user_permissions/<id>", methods=["GET"])
+@with_session
+def get_user_permissions(id):
+    """
+
+        Checks to see if user has permissions to access a certain user.
+
+    """
+    try:
+        user = User.query.filter_by(id=id).one()
+        return get_class_permissions(user.cohort_id)
+    except:
+        flask.abort(401)
+
+
+# Asking for a nonexistant cohort will cause .one() to crash!
+# Takes cohort_id and returns all users belonging to that cohort
+@api.route("/get_users_from_class/<id>", methods=["GET"])
+@with_session
+def get_users_from_class(id):
+    if (not _has_permission_for_class(id)):
+        flask.abort(401)
+    try:
+        c = Cohort.query.filter_by(id=id).one()
+        users = User.query.filter_by(cohort_id=c.id).all()
+        users_info = []
+        for u in users:
+            info = _get_user_info(u.id)
+            users_info.append(info)
+        return json.dumps(users_info)
+
+    except:
+        flask.abort(400)
+
+
+# Takes cohort_id and reuturns dictionary with relevant class variables
+@api.route("/get_user_info/<id>", methods=["GET"])
+@with_session
+def wrapper_to_json_user(id):
+    if (not get_user_permissions(id)):
+        flask.abort(401)
+    return jsonify(_get_user_info(id))
+
+
+# Gets user info
+def _get_user_info(id):
+    try:
+        user = User.query.filter_by(id=id).one()
+        dictionary = {
+            'id': str(id),
+            'name': user.name,
+            'email': user.email,
+            'reading_time': random.randint(1, 100),
+            'exercises_done': random.randint(1, 100),
+            'last_article': 'place holder article'
+        }
+        return dictionary
+    except ValueError:
+        flask.abort(400)
+
+
+# Removes class.Can only be called successfuly if the class is empty. Otherwise we return 400 (Value error).
+@api.route("/remove_class/<class_id>", methods=["POST"])
```

Comment:
ok, sorry, if i was undecided in the previous feedback round.
let's stop talking about classes, and use only `cohort` instead. 
if everywhere else in the project we talk about cohorts, it is 
unprofessional to talk only here about 'classes'. 

so please, let's refactor this to cohort.

Reply:
Do you want me to also rename the variables in cohort from class_name to cohort_name etc?
This might take me a while to do as it requires changes from many functions in the API and the WEB

Reply:
Sure. Nice idea. Or name it only "name" given that it.s already scoped in cohort?

Reply:
All API funcitons have been renamed accordingly. All variable names in Cohort have been renamed accordingly. All tests have been updated to accommodate for changes. I might save the function names as variables in testing in future to make test updating faster.